### PR TITLE
Order Creation: Add basic payment section

### LINKED ISSUE #5411: Order Creation: Basic Payment details section
For milestone 1, we want to display a basic, read-only Payment section (no buttons to add payment details). This should appear after at least one product is added to the order, and should include the Products Total and Order Total.

Ref: p91TBi-6xc-p2#comment-7236 

### Tasks

- [ ] UI for Payment section on New Order screen with read-only Products Total and Order Total

### PR DESCRIPTION
**Pull request by rachelmcr**
Closes: #5411

## Description

This adds a basic Payment section to the new order screen, in the order creation flow.

The Payment section appears when there is at least one product in the order. For now, it is a read-only summary showing the total cost of the products in the order.

**Question:** Should we show some kind of info text about the "Order Total" to explain that taxes may be calculated automatically and added to this total? (Also raised in Slack: p1641317781008900-slack-C02KUCFCSFP)

## Changes

* New `OrderPaymentSection` view that displays the Products Total and Order Total with formatted amounts.
* Adds `OrderPaymentSection` to `NewOrder` view.
* `NewOrderViewModel` manages the Payment section properties:
   * `shouldShowPaymentSection` indicates if the Payment section should be shown, depending on whether there are items in the order.
   * `PaymentDataViewModel` is a new struct that represents the payment data. It takes unformatted totals and formats them according to the provided currency formatter.
   * `configurePaymentDataViewModel()` updates the payment data based on the total price of the items in the order.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab, tap the + button, and select "Create order."
4. Confirm there is no Payment section visible on the New Order screen.
5. Add a product to the order.
6. Confirm a Payment section now appears with the correct product total.
7. Change the products in the order (multiple products, different quantities) and confirm the Payment section updates accordingly.

## Screenshots

https://user-images.githubusercontent.com/8658164/148113270-0245b762-65a0-4085-b7c2-a03f874719f2.mp4

## Submitter Checklist

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/OrderPaymentSection.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    let viewModel: NewOrderViewModel.PaymentDataViewModel
+
+    ///   Environment safe areas
+    ///
+    @Environment(\.safeAreaInsets) var safeAreaInsets: EdgeInsets
+
+    var body: some View {
+        Divider()
+
+        VStack(alignment: .leading, spacing: Layout.verticalSpacing) {
```
Comment (ealeksandrov) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/OrderPaymentSection.swift]: nit: I think 0 spacing is better to be used as inline value. Or maybe have it named obviously in constants, like `zeroSpacing`. Or use `.zero` of CGFloat.
Its meaning here is more like "no value" and having it inline helps improve readability.

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift [hunk 2]
```diff
@@ -60,6 +60,13 @@ struct NewOrder: View {
 
                         Spacer(minLength: Layout.sectionSpacing)
 
+                        Group {
+                            OrderPaymentSection(viewModel: viewModel.paymentDataViewModel)
+
+                            Spacer(minLength: Layout.sectionSpacing)
+                        }
+                        .renderedIf(viewModel.shouldShowPaymentSection)
```
Comment (ealeksandrov) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift]: nit: when we have to add Group to use `renderedIf` - isn't it simpler to just write `if x { }`? 🙂

**Review by ealeksandrov — APPROVED** [review]
LGTM, just couple of nits/questions :shipit:

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/OrderPaymentSection.swift (hunk 1 above)
Comment (rachelmcr) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/OrderPaymentSection.swift]: Oh, agreed, thanks for catching that! I had tried it out with more spacing and forgot that I'd set to to 0 in the end.

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift (hunk 2 above)
Comment (rachelmcr) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift]: True :)

**Comment by rachelmcr** [comment]
Thanks for the review! I also added info text about the taxes in d614bf6, based on the conversation in Slack (ref: p1641317781008900-slack-C02KUCFCSFP). We can remove it once we support taxes in new orders:

**Pull request merged**